src/Vector.cpp

```diff
@@ -0,0 +1,436 @@
+            res += abs(vals[i]);
+        break;
+
+    case NORM_2:
+        for (unsigned int i = 0; i < size; ++i)
+            res += vals[i] * vals[i];
+        res = sqrt(res);
+        break;
+
+    case NORM_INF:
+        for (unsigned int i = 0; i < size; ++i)
+        {
+            if (abs(vals[i]) > res)
+                res = abs(vals[i]);
+        }
+        break;
+
+    default:
+        ILog::report("IVector.norm: Norm not defined.\n");
+        return ERR_NORM_NOT_DEFINED;
+    }
+    return ERR_OK;
+}
+
+int Vector::setCoord(unsigned int index, double elem)
+{
+    RANGE_CHECK(index, "setCoord");
+    vals[index] = elem;
+
+    return ERR_OK;
+}
+
+int Vector::getCoord(unsigned int index, double & elem) const
+{
+    RANGE_CHECK(index, "getCoord");
+    elem = vals[index];
+
+    return ERR_OK;
+}
+
+int Vector::setAllCoords(unsigned int dim, double* coords)
+{
+    if (dim != size)
+    {
+        ILog::report("IVector.setAllCoords: Dimensions mismatch.\n");
+        return ERR_DIMENSIONS_MISMATCH;
+    }
+
+    if (!coords)
+    {
+        ILog::report("IVector.setAllCoords: Coords - nullptr.\n");
+        return ERR_WRONG_ARG;
+    }
+
+    for (unsigned int i = 0; i < size; ++i)
+        vals[i] = coords[i];
+
+    return ERR_OK;
+}
+
+int Vector::getCoordsPtr(unsigned int & dim, double const*& elem) const
+{
+    dim = size;
+    elem = vals;
+
+    return ERR_OK;
+}
+
+IVector* Vector::clone() const
+{
+    return createVector(size, vals);
+}
+
+int Vector::gt(IVector const* const right, NormType type, bool& result) const
+{
+    if (!right)
+    {
+        ILog::report("IVector.gt: Right - nullptr.\n");
+        return ERR_WRONG_ARG;
+    }
+
+    double norm, normR;
+    int errCode;
+
+    errCode = this->norm(type, norm);
+    if (errCode != ERR_OK)
+        return errCode;
+    errCode = right->norm(type, normR);
+    if (errCode != ERR_OK)
+        return errCode;
+
+    result = norm > normR;
+
+    return ERR_OK;
+}
+
+int Vector::lt(IVector const* const right, NormType type, bool& result) const
+{
+    if (!right)
+    {
+        ILog::report("IVector.lt: Right - nullptr.\n");
+        return ERR_WRONG_ARG;
+    }
+
+    double norm, normR;
+    int errCode;
+
+    errCode = this->norm(type, norm);
+    if (errCode != ERR_OK)
+        return errCode;
+    errCode = right->norm(type, normR);
+    if (errCode != ERR_OK)
+        return errCode;
+
+    result = norm < normR;
+
+    return ERR_OK;
+}
+
+int Vector::eq(IVector const* const right, NormType type, bool& result, double precision) const
+{
+    if (!right)
+    {
+        ILog::report("IVector.eq: Right - nullptr.\n");
+        return ERR_WRONG_ARG;
+    }
+
+    IVector *v = IVector::subtract(this, right);
+    if (!v)
+    {
+        ILog::report("IVector.eq: Can't subtract vectors.\n");
+        return ERR_WRONG_ARG;
+    }
+
+    double norm;
+    int errCode = v->norm(type, norm);
+    if (errCode != ERR_OK)
+        return errCode;

```

Comment:
It needs to delete `v` in this case. You can delete this after 
```c++
    int errCode = v->norm(type, norm);
```
and erase 
```c++
    delete v;
```
on line 428.